lib/pbench/server/api/resources/__init__.py

```diff
@@ -1457,11 +1457,14 @@ def _get_dataset_metadata(
             if Metadata.is_key_path(i, Metadata.METADATA_KEYS):
                 native_key = Metadata.get_native_key(i)
                 user: Optional[User] = None
+                user_id = None
                 if native_key == Metadata.USER:
                     user = Auth.token_auth.current_user()
+                    if user:
+                        user_id = user.id
                 try:
                     metadata[i] = Metadata.getvalue(
-                        dataset=dataset, key=i, user_id=user.id if user else None
+                        dataset=dataset, key=i, user_id=user_id
                     )
```

Comment:
`user` is only used at lines 1462-1463, so you should "declare" it in that block, or just drop line 1459.

Ditto `lib/pbench/server/api/resources/datasets_metadata.py` line 179.

Reply:
Maybe it's better to just drop it (anyway it's a `User` field and will get dropped eventually)

Reply:
You mean, just assign `user_id` directly from `Auth.token_auth.current_user()` instead of saving `user`? Yeah, maybe.

Reply:
I'm lost....  We need some notion of the user's identity (if there is one) at line 1467.  That notion comes from the token at line 1462 (if we're dealing with a `USER` metadata key).  Currently, "the user" is a username which we convert to an "id" value at line 1464; someday that conversion will presumably be done by Keycloak, and the "id" will be a UUID instead of a row number, but that's not important right now.

So, I don't see how we can drop the notion of a user from this sequence.  We could try to extract the `id` directly from the `Auth` call, but I think that `current_user()` can return `None` which is going to make that awkward.

Anyway, I think the code is good the way it is...except for the pointless assignment at line 1459.  If we going to drop something, it should be just that line.  🙂